py.test test discovery

Hi there, I'm having a problem with getting py.test and freezegun to play well together. I'm not sure which of the two is to blame or whether I'm doing something wrong here, but I was hoping you might be able to shed some light on this.

All of this is using freezegun 0.1.18 from PyPi and py.test 2.6.1.

The simplest test works fine:

``` python
# test_discovery.py
class TestDiscovery(object):
    def test(self):
        assert True
```

``` plaintext
> py.test test_discovery.py
========================== test session starts ==========================
platform darwin -- Python 2.7.5 -- py-1.4.23 -- pytest-2.6.1
collected 1 items

test_discovery.py .
```

The test is found and run, all great. But when I add freezegun, the test is no longer being found:

``` python
from freezegun import freeze_time

@freeze_time("2014-08-05")
class TestDiscovery(object):
    def test(self):
        assert True
```

``` plaintext
> py.test test_discovery.py
========================== test session starts ==========================
platform darwin -- Python 2.7.5 -- py-1.4.23 -- pytest-2.6.1
collected 0 items
```

I'm running into similar (but possibly unrelated) issues when I want to freeze time on a function that uses a py.test fixture (this test works fine if I leave out freezegun):

``` python
import pytest
from freezegun import freeze_time

@pytest.fixture
def fixture():
    return 42

@freeze_time("2014-08-05")
def test(fixture):
    assert fixture == 42
```

```
> py.test test_discovery.py
========================== test session starts ==========================
platform darwin -- Python 2.7.5 -- py-1.4.23 -- pytest-2.6.1
collected 1 items

test_discovery.py F

=============================== FAILURES ================================
_________________________________ test __________________________________

args = (), kwargs = {}

    def wrapper(*args, **kwargs):
        with self:
>           result = func(*args, **kwargs)
E           TypeError: test() takes exactly 1 argument (0 given)

venv/lib/python2.7/site-packages/freezegun/api.py:219: TypeError
```

Any hints or ideas are much appreciated! Thanks.

--- a/freezegun/api.py
+++ b/freezegun/api.py
@@ -292,6 +292,11 @@ class _freeze_time(object):
                 result = func(*args, **kwargs)
             return result
         functools.update_wrapper(wrapper, func)
+
+        # update_wrapper already sets __wrapped__ in Python 3.2+, this is only
+        # needed for Python 2.x support
+        wrapper.__wrapped__ = func
+
         return wrapper
 
 

--- a/tests/test_datetimes.py
+++ b/tests/test_datetimes.py
@@ -161,6 +161,14 @@ def test_decorator():
     assert datetime.datetime.now() == datetime.datetime(2012, 1, 14)
 
 
+def test_decorator_wrapped_attribute():
+    def to_decorate():
+        pass
+
+    wrapped = freeze_time("2014-01-14")(to_decorate)
+
+    assert wrapped.__wrapped__ is to_decorate
+
 @freeze_time("2012-01-14")
 class Tester(object):
     def test_the_class(self):

